urlchecker/core/urlproc.py

```diff
@@ -146,13 +149,100 @@ def extract_urls(self):
         # collect all links from file (unique=True is set)
         self.urls = fileproc.collect_links_from_file(self.file_name)
 
-    def check_urls(self, urls=None, retry_count=1, timeout=5):
+    async def aysnc_url_check(self, url, retry_count, timeout, headers):
+        """
+        Asyncronous check function for one url.
+
+        Args:
+            - urls           (list) : a list of urls to check.
+            - retry_count    (int)  : a number of retries to issue (defaults to 1, no retry).
+            - timeout        (int)  : a timeout in minutes for blocking operations like the connection attempt.
+            - headers        (dict) : headers to use in the request.
+        """
+        # init do retrails and retrails counts
+        do_retry = True
+        rcount = retry_count
+
+        # we will double the time for retry each time
+        retry_seconds = 15
+
+        # With retry, increase timeout by a second
+        pause = timeout
+        saved_responses = []
+        saved_errors = []
+
+        try:
```

Comment:
It would be best to not put this entire thing in a try / except, and figure out where/what errors are thrown and catch them appropriately.